src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -155,22 +155,100 @@ public User getUser(String email) {
     return user;
   }
 
-    /**
-     * Gather a list of all users that have entered something into the community chat.
-     *
-     * @return a Set of users, no order to this list, empty list if there are no users that have posted
-     */
-    public Set<String> getUsers () {
-      Set<String> users = new HashSet<>();
-      Query query = new Query("Message");
-      PreparedQuery results = datastore.prepare(query);
-      for (Entity entity : results.asIterable()) {
-        users.add((String) entity.getProperty("user"));
+  /**
+   * Gather a list of all users that have entered something into the community chat.
+   *
+   * @return a Set of users, no order to this list, empty list if there are no users that have posted
+   */
+  public Set<String> getUsers () {
+    Set<String> users = new HashSet<>();
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      users.add((String) entity.getProperty("user"));
+    }
+    return users;
+
+  }
+
+  /** Stores a new  Place in Datastore. */
+  public void storePlace(Place place) {
+    Entity placeEntity = new Entity("Place", place.getId().toString());
+    placeEntity.setProperty("owner", place.getOwner());
+    placeEntity.setProperty("title", place.getTitle());
+    placeEntity.setProperty("description", place.getDescription());
+    placeEntity.setProperty("latitude", place.getLatitude());
+    placeEntity.setProperty("longitude", place.getLongitude());
+    placeEntity.setProperty("timestamp", place.getTimestamp());
+
+    datastore.put(placeEntity);
+  }
+
+  /** Get all the messages currently in the Datastore. */
+  public List<Place> getAllPlaces() {
+    List<Place> places = new ArrayList<>();
+    Query query = new Query("Place")
+        .addSort("timestamp", SortDirection.DESCENDING);
+
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      try {
+        String idString = entity.getKey().getName();
+        UUID id = UUID.fromString(idString);
+        String owner = (String) entity.getProperty("owner");
+        String title = (String) entity.getProperty("title");
+        String description = (String) entity.getProperty("description");
+        long latitude = (long) entity.getProperty("latitude");
+        long longitude = (long) entity.getProperty("longitude");
+        long timestamp = (long) entity.getProperty("timestamp");
+        Place place = new Place(id, owner, title, description, latitude, longitude, timestamp);
+        places.add(place);
+      } catch (Exception e) {
```

Comment:
Please catch a more granular exception; catching Exception is a bad practice.

Reply:
This is based on old code that passed code review.

Reply:
Didn't know that it was from CodeU, but could you properly fix it? It's easy to find out which specific exceptions to catch by removing the try catch and seeing what exceptions Java says the method must throw.

Reply:
okay so i tried finding the specific exception but didn't get any maven/console prompts when removing the catch block. also yea it's following the same pattern codeu code, so i think if there was specific exceptions to catch they wouldve added them since they did for other files.